### DISCUSSION TITLE
Functionality to add to COBREXA

### DISCUSSION THREAD

**Discussion by stelmo** [LCSB-BioCore/COBREXA.jl#283] [Ideas]
Here are some potential features/improvements to add:

- [ ] Flux coupling analysis
- [ ] Eigenreaction analysis (this is great for GPUs!)
- [ ] Model reduction, e.g. combine reactions that are stoichiometrically directly coupled
- [ ] dynamic flux balance analysis, I have reservations about this since dFBA is tricky once you start thinking about it
- [ ] Data integration (very broad)
- [x] Improve model reading performance, especially json and sbml 
- [ ] Community visualization, e.g. which exchanges are connected to which organisms in the community?
- [x] Model exploration functions, e.g. something that simplifies `findall(x -> occursin("biomass", x), reactions(model))` for example. This is most useful if you loaded a model but know nothing about it (e.g. an automatically generated one). Another example, find all the reactions directly linked to metabolite `X` etc.
- [ ] Convergence estimation for sampling - ~~tricky~~ borderline impossible with high dimensionality, will require some thought 🤔 
- [ ] Growth balance analysis
- [x] Moment + Moma combo?

Anything else @laurentheirendt @htpusa @exaexa?

**Comment by htpusa** [comment]
Elementary Conversion Modes could be interesting. Originally from [1] but more recently in [2]. See especially the discussion about scaling up the method in the latter.

1. [Urbanczik, R., & Wagner, C. (2005). Functional stoichiometric analysis of metabolic networks. Bioinformatics, 21(22), 4176-4180.]( https://doi.org/10.1093/bioinformatics/bti674)
2. [Clement, T. J., Baalhuis, E. B., Teusink, B., Bruggeman, F. J., Planqué, R., & de Groot, D. H. (2021). Unlocking Elementary Conversion Modes: ecmtool unveils all capabilities of metabolic networks. Patterns, 2(1), 100177.](https://doi.org/10.1016/j.patter.2020.100177)

**Reply by exaexa** [reply]
if I get it correctly the ECMs are basically eigenvectors in the optimum space that is cut down to exchange reactions only, right?

**Comment by laurentheirendt** [comment]
The following should be there I think:

dFBA: https://pubmed.ncbi.nlm.nih.gov/12202358/
MOMA: minimization of metabolic adjustment

**Reply by stelmo** [reply]
So with respect to dFBA, we should implement this algorithm: https://doi.org/10.1002/bit.24748
However, I have serious doubts about dFBA approaches in general, but it is very popular so I guess we should add it :/

**Comment by exaexa** [comment]
I added some minor ideas (esp. the gene-moma from yesterday, can we call it momament?)

Question 1: "Flux coupling" is the same thing as in CoreModelCoupled right? (Can't we make a "wrapper" coupler model that could work like `CoupledModel{StandardModel}`? )

Question 2: Eigenreactions is really just finding the eigenvalues right?

**Reply by exaexa** [reply]
Nice data here:  https://www.embopress.org/doi/full/10.15252/msb.20209536

**Reply by htpusa** [reply]
1) Yes. And yes, I agree that something other than a separate type would make sense.
Although what one really needs is just adding inequality constraints which could just be a generic functionality.

2) "Eigenreactions" comes from [here](https://www.sciencedirect.com/science/article/pii/S0022519303001462). It's basically an interpretation of the singular value decomposition of the stoichiometric matrix.

**Reply by exaexa** [reply]
o nice thanks!
